Library/Homebrew/diagnostic.rb

```diff
@@ -304,11 +304,19 @@ def check_for_broken_symlinks
       end
 
       def check_for_unsupported_osx
-        if !ARGV.homebrew_developer? && OS::Mac.prerelease? then <<-EOS.undent
-        You are using OS X #{MacOS.version}.
-        We do not provide support for this pre-release version.
-        You may encounter build failures or other breakages.
-        EOS
+        return if ARGV.homebrew_developer?
+        if OS::Mac.prerelease?
+          <<-EOS.undent
+            You are using OS X #{MacOS.version}.
+            We do not provide support for this pre-release version.
+            You may encounter build failures or other breakages.
+          EOS
+        elsif OS::Mac.version < :mavericks
```

Comment:
How about adding a new method (e.g. `OS::Mac#outdated_release?`) similar to `OS::Mac#prerelease?`